### DISCUSSION TITLE
Moving from Deconz to Zigbee2Mqtt WITHOUT re-pairing all devices?

### DISCUSSION THREAD

**Discussion by wexman** [Koenkk/zigbee2mqtt#10241] [General]
Hi, I'm using a ConBee II adapter and want to move from Deconz zo Zigbee2Mqtt because I think it fits better for my needs.

Will this be possible without re-pairing all devices if I keep the same pan id, channel and network key? (if these infos are readable from deconz in any way)?

Thanks for any input!

**Comment by Koenkk** [comment]
This is not possible

**Comment by wexman** [comment]
Thanks. That is a general zigbee limitation? Or a Deconz? Or even zigbee2mqtt? Or is my idea just stupid? In the latter case I apologize, I'm quite new to this...

**Reply by Koenkk** [reply]
It is a software limitation, no conversion tools are available.

**Reply by wexman** [reply]
I wasn't expecting a fully automated conversion in any way!
What I meant was if it was somehow possible to configure - manually - ZigBee2mqtt with the same id's and keys that deconz is using so I could avoid having to re-pair the devices...

**Reply by samsam-rolon** [reply]
Yeah ! I was intrested too https://github.com/Koenkk/zigbee2mqtt/discussions/6230
I have fully intregated legrand system so re-paring every devices is just impossible for me ... A method, not automated but software based (and not based on me unmounting all my device to use pairing button) will be pleasant :-)

**Reply by sretalla** [reply]
My experience after having done this...

Most of the lights came over with the conbee II stick and didn't need re-pairing (although did need to be powered off and on again in most cases). They did, however, need naming, so almost as difficult as pairing and took ages and a lot of care.

Almost all of my Aqara Sensors needed pairing again after the switch. Almost none of the Philips sensors needed to be re-paired, but all needed naming (again complex to figure out as just one person when they are all over the house).

What I needed in order to make the switch:
Go to the Network side (not the Phoscon App of the Add-on) and have a look under Edit | Network

Note down the PAN ID and EXT PAN ID (I note I wasn't able to get zigbee2mqtt to take this value no matter what I did, but it seemed not to matter in the end) and note your network key and TC Link Key

I also converted my network key from the hex value to the 16 numbers (2 characters at a time after the 0x) fir using in the add-on config network key value... not sure if this was used by the add-on in the end as it's also in the coordinator backup)

Then stop the add-on and fill those into a file in your zigbee2mqtt directory called coordinator_backup.json (you'll need to have already started the add-on with initial settings for this file to have been created).

Start the add-on and go see what you have.

I was subsequently able to just change over to a new stick (not conbee, rather Sonoff USB3) and everything just came over... a couple of devices needed a power cycle, naming was kept.

**Reply by MTrab** [reply]
> fill those into a file in your zigbee2mqtt directory called coordinator_backup.json (you'll need to have already started the add-on with initial settings for this file to have been created)

Do I need to do anything else to have the file created?
Have started Z2M several times, but the file is not generated

**Reply by sretalla** [reply]
It's possible that I had tested pairing one device with the new adapter which might have been the thing that made it create the file.

**Reply by srd424** [reply]
Hmm, I've just done this and didn't get the coordinator backup created either (and I tried properly repair one or two devices as well, and that didn't trigger it.) Most of the lights and sensors have come across though, and it turns out I had a script buried somewhere which had the sensor MAC addresses buried in it, so I've got a bit of a head start on the naming :)

**Reply by Hedda** [reply]
dresden-elektronik deCONZ developers need to add an option to save backups in the "Open ZigBee Coordinator Backup Format" standard:

Check out the directly related feature request Dresden Elektronik here ->  https://github.com/dresden-elektronik/deconz-rest-plugin-v2/issues/12

**Reply by lanwin** [reply]
@sretalla thank you for that tutorial. You mentioned that you have written down the TC Link Key. I did not see a way in z2m to put that in.

Also I see my devices came in agian. But the HUE devices show "interview failed".

**Comment by Hedda** [comment]
dresden-elektronik deCONZ developers needs to add an option to save backups in the "Open ZigBee Coordinator Backup Format" standard:

Check out this feature request Dresden Elektronik ->  https://github.com/dresden-elektronik/deconz-rest-plugin-v2/issues/12

**Comment by WegnerDan** [comment]
> I also converted my network key from the hex value to the 16 numbers (2 characters at a time after the 0x) fir using in the add-on config network key value... not sure if this was used by the add-on in the end as it's also in the coordinator backup)

Could you elaborate how you converted the network key?
As fair as I can tell, the 0x prefix just needs to be removed. Am I missing something?

**Comment by Prior99** [comment]
Is the current state still that it is not possible to migrate an existing network to Z2M?

I think all the values that are needed are within the coordinator's Network Settings:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1491574/fabe7f68-68c5-4028-97ee-7cef93b54a46)

Wouldn't it be possible to join the coordinator into the existing network and have two coordinators until the migration is complete?

I am asking because I have some Zigbee devices for controlling window blinds and they are inside the wall. Re-pairing them would mean that I have to spend at least an hour for each of them with opening everything up, disconnecting power, etc.

Is there no way at all of migrating without re-pairing everything? I don't really care about renaming and configuring my 159 devices manually, it's just the process of pairing that I am afraid of.  It is also possible to load a backup file from the deconz controller that looks like this:

```
{
  "apsAck": false,
  "apsUseExtPanId": "0x0",
  "curChannel": 11,
  "deconzVersion": "22102",
  "deviceType": 0,
  "endpoint1": {
    "deviceId": "0x5",
    "deviceVersion": "0x1",
    "endpoint": "0x1",
    "inClusters": [
      "0x0",
      "0x6",
      "0xa",
      "0x19",
      "0x501"
    ],
    "index": 0,
    "outClusters": [
      "0x1",
      "0x20",
      "0x500",
      "0x502"
    ],
    "profileId": "0x104"
  },
  "endpoint2": {
    "deviceId": "0x64",
    "deviceVersion": "0x1",
    "endpoint": "0xf2",
    "inClusters": [],
    "index": 1,
    "outClusters": [
      "0x21"
    ],
    "profileId": "0xa1e0"
  },
  "extPanId": "0x2[...]0",
  "frameCounter": 218007394,
  "macAddress": "0x2[...]90",
  "networkKey": "3fc4[...]a5",
  "nwkAddress": "0x0",
  "nwkUpdateId": 4,
  "otauactive": 1,
  "panId": "0x6[...]b",
  "securityMode": 3,
  "staticNwkAddress": false,
  "tcAddress": "0x21[...]90",
  "tcLinkKey": "5a69[...]39"
}
```

No chance?

**Reply by Hedda** [reply]
No, above statement still applies:

> dresden-elektronik deCONZ developers needs to add an option to save backups in the "Open ZigBee Coordinator Backup Format" standard:
> 
> Check out this feature request Dresden Elektronik -> https://github.com/dresden-elektronik/deconz-rest-plugin-v2/issues/12

DeCONZ/Phoscon backup format is propriatory and is not compatible with any third-party Zigbee Gateway applications.

**Comment by Hedda** [comment]
> I am asking because I have some Zigbee devices for controlling window blinds and they are inside the wall. Re-pairing them would mean that I have to spend at least an hour for each of them with opening everything up, disconnecting power, etc.

When you remove a device from a Zigbee network then it will usually put that device into pairing/joining mode for around 1-minute, so if you are ready and put Zigbee2MQTT to allow pairing/joining mode at the time it is removed then if you are lucky it should hopefully automatically pair/join, (but note that not all Zigbee devices are automatically put into pairing/joining mode when removed).

**Reply by Prior99** [reply]
I tried this with some light bulbs from innr and hue and it did not work. The bulbs did not enter joining mode after I removed them from the network.

I am currently writing a script to migrate from the deconz json file and sqlite database to z2m, however the new stick can not join the network:

```
Zigbee2MQTT:info  2024-04-16 09:43:45: Logging to console and directory: '/home/prior/repositories/zigbee2mqtt/data/log/2024-04-16.09-43-45' filename: log.txt
Zigbee2MQTT:info  2024-04-16 09:43:45: Starting Zigbee2MQTT version 1.36.1 (commit #ffc2ff1)
Zigbee2MQTT:info  2024-04-16 09:43:45: Starting zigbee-herdsman (0.40.3)
Zigbee2MQTT:error 2024-04-16 09:44:53: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-04-16 09:44:53: Failed to start zigbee
Zigbee2MQTT:error 2024-04-16 09:44:53: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-04-16 09:44:53: Exiting...
Zigbee2MQTT:error 2024-04-16 09:44:53: Error: network commissioning timed out - most likely network with the same panId or extendedPanId already exists nearby
    at ZnpAdapterManager.beginCommissioning (/home/prior/repositories/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:347:23)
    at ZnpAdapterManager.start (/home/prior/repositories/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/manager.ts:92:17)
    at Controller.start (/home/prior/repositories/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/home/prior/repositories/zigbee2mqtt/lib/zigbee.ts:62:27)
    at Controller.start (/home/prior/repositories/zigbee2mqtt/lib/controller.ts:109:27)
    at start (/home/prior/repositories/zigbee2mqtt/index.js:107:5)
```

I tried to change the stick's IEEE address to the MAC address the old Conbee II stick has and unplugged the old stick, but the new stick will not accept the IEEE address.

**Reply by lanwin** [reply]
@Hedda thank you for that! It worked very well.

For me it worked for Hue, Osram and Ikea but not for the Aqura Sensors.

**Reply by BeamMeUpTo** [reply]
Perhaps the solution for the topic is this..... 

https://github.com/chr-braun/deconz-to-zigbee2mqtt

It is not a coordinator change from other device than conbee/raspbee to X, but deconz to zigbee2mqtt.

Perhaps helpful
